src/fender_value.rs

```diff
@@ -353,6 +400,69 @@ impl ToString for FenderValue {
             ),
             FenderValue::Struct(s) => s.to_string(),
             FenderValue::Type(t) => format!("Type({})", t.to_string()),
+            FenderValue::HashMap(h) => format!(
+                "[{}]",
+                h.iter()
+                    .map(|(k, v)| format!(
+                        "{}:{}",
+                        k.to_literal_display_string(),
+                        v.to_literal_display_string()
+                    ))
+                    .collect::<Vec<_>>()
+                    .join(", ")
+            ),
+        }
+    }
+}
+
+impl PartialEq for FenderValue {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Ref(a), Self::Ref(b)) => a == b,
+            (Self::Int(a), Self::Int(b)) => a == b,
+            (Self::Float(a), Self::Float(b)) => a == b,
+            (Self::Char(a), Self::Char(b)) => a == b,
+            (Self::String(a), Self::String(b)) => a == b,
+            (Self::Bool(a), Self::Bool(b)) => a == b,
+            (Self::Error(a), Self::Error(b)) => a == b,
+            (Self::Function(a), Self::Function(b)) => a == b,
+            (Self::List(a), Self::List(b)) => a == b,
+            (Self::Struct(a), Self::Struct(b)) => a == b,
+            (Self::Type(a), Self::Type(b)) => a == b,
+            (Self::HashMap(a), Self::HashMap(b)) => a == b,
+            _ => core::mem::discriminant(self) == core::mem::discriminant(other),
```

Comment:
I'm not really a fan of using `_ =>` here, since it means that any time a new type is added, we will need to remember to update this function. It'll be *extremely* easy to forget and could cause bugs.

Reply:
this is the default implementation that was generated, we have to use it or implement every permutation of this.
we could change it to panic instead tho

Reply:
Hm, I just wish there was a good way for us to get a compiler error when a new type is added forcing us to update this function

Reply:
nvm this can't panic it has to be as it is because the hashmap will make checks against its keys and if they aren't the same type they would cause the panic

Reply:
Yeah... I'm really trying to think of a way around it, that would force us to implement equality for new types. Maybe a macro? I can't think of how that macro would work though, unless it could generate all permutations of types - and I'm not sure that would be a good idea even if we could do it.